File: uphsd-cs-coloma-aysbergmatt/MIDTERM WORK 2/Dataset.js
/**
 * Student Name: COLOMA, AYSBERG MATT A.
 * Machine Problems: MP01, MP13, MP19
 */

const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter the dataset file path (e.g., MOCK_DATA.csv): ', (filePath) => {
    try {
        const fileContent = fs.readFileSync(filePath, 'utf8');
        const lines = fileContent.split(/\r?\n/).filter(line => line.trim() !== "");
        
        if (lines.length === 0) {
            console.log("Error: File is empty.");
            rl.close();
            return;
        }

        // CSV Regex to handle commas inside quotes
        const csvSplitBy = /,(?=(?:(?:[^"]*"){2})*[^"]*$)/;

        const headers = lines[0].split(csvSplitBy).map(h => h.trim().replace(/"/g, ""));
        const records = lines.slice(1).map(line => line.split(csvSplitBy));

        // Execute Machine Problems
        displayMP01(records);
        const missingRows = findMP13(records);
        displayMP19(records, missingRows);

    } catch (err) {
        console.log("Error: " + err.message);
    } finally {
        rl.close();
    }
});

// [MP01] Basic record count
function displayMP01(data) {
    console.log(`\n[MP01] Total Records: ${data.length}`);
}

// [MP13] Logic to find rows with empty/null fields
function findMP13(data) {
    let rowsWithMissing = [];
    data.forEach((row, index) => {
        const hasMissing = row.some(field => !field || field.trim() === "" || field.trim() === '""');
        if (hasMissing) {
            rowsWithMissing.push(index + 1);
        }
    });
    console.log(`[MP13] Rows with missing values: [${rowsWithMissing.join(", ")}]`);
    return rowsWithMissing;
}

// [MP19] Summary Report
function displayMP19(data, missingRows) {
    console.log("\n--- [MP19] DATASET SUMMARY REPORT ---");
    console.log(`Total Records:   ${data.length}`);
    console.log(`Incomplete Rows: ${missingRows.length}`);
    console.log(`Data Integrity:  ${data.length - missingRows.length} complete rows.`);
    console.log("-------------------------------------");
}

File: uphsd-cs-coloma-aysbergmatt/MIDTERM WORK 2/Dataset.java
/**
 * Student Name: COLOMA, AYSBERG MATT A.
 * Machine Problems: MP01, MP13, MP19
 */

import java.io.*;
import java.util.*;

public class Dataset {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the dataset file path (e.g., MOCK_DATA.csv): ");
        String filePath = input.nextLine();
        
        List<String[]> records = new ArrayList<>();
        String[] headers = null;

        // Regex to handle commas inside quoted strings
        String csvSplitBy = ",(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)";

        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
            String line = br.readLine();
            if (line != null) {
                headers = line.split(csvSplitBy, -1);
            }

            while ((line = br.readLine()) != null) {
                records.add(line.split(csvSplitBy, -1));
            }

            displayMP01(records);
            List<Integer> missingRows = findMP13(records, headers);
            displayMP19(records, missingRows);

        } catch (IOException e) {
            System.out.println("Error: " + e.getMessage());
        } finally {
            input.close();
        }
    }

    // [MP01] Basic record count
    public static void displayMP01(List<String[]> data) {
        System.out.println("\n[MP01] Total Records: " + data.size());
    }

    // [MP13] Logic to find rows with empty/null fields
    public static List<Integer> findMP13(List<String[]> data, String[] headers) {
        List<Integer> rowsWithMissing = new ArrayList<>();
        for (int i = 0; i < data.size(); i++) {
            for (String field : data.get(i)) {
                if (field == null || field.trim().isEmpty()) {
                    rowsWithMissing.add(i + 1);
                    break;
                }
            }
        }
        System.out.println("[MP13] Rows with missing values: " + rowsWithMissing);
        return rowsWithMissing;
    }

    // [MP19] Summary Report - Now focused only on record totals and integrity
    public static void displayMP19(List<String[]> data, List<Integer> missingRows) {
        System.out.println("\n--- [MP19] DATASET SUMMARY REPORT ---");
        System.out.println("Total Records:   " + data.size());
        System.out.println("Incomplete Rows: " + missingRows.size());
        System.out.println("Data Integrity:  " + (data.size() - missingRows.size()) + " complete rows.");
        System.out.println("-------------------------------------");
    }
}